other/upgrade_2-1_postgresql.sql

```diff
@@ -2308,18 +2308,42 @@ DROP INDEX IF EXISTS {$db_prefix}topics_id_board;
 --- update ban ip with ipv6 support
 /******************************************************************************/
 ---# add columns
-ALTER TABLE {$db_prefix}ban_items ADD COLUMN ip_low inet;
-ALTER TABLE {$db_prefix}ban_items ADD COLUMN ip_high inet;
+---{
+$request = upgrade_query("
+	SELECT column_name
+	FROM INFORMATION_SCHEMA.COLUMNS WHERE table_name = '{$db_prefix}ban_items';
+");
+$upcontext['ban_items_columns'] = $smcFunc['db_fetch_all']($request);
+$smcFunc['db_free_result']($request);
+
+if (!in_array('ip_low', $upcontext['ban_items_columns']))
```

Comment:
the better code would be
```
$smileys_columns = $smcFunc['db_list_columns']('{$db_prefix}ban_items');
if (in_array('ip_low', $smileys_columns))
{
```

Reply:
I know it's less code to do it that way, but it seems a bit silly to go through the DB abstraction layer when we already know what DB system we're using.

Reply:
It's less code and the smf default way to retrieve the information,
the sql command which sequi use is very abnomarl way in pg world to retrieve the information.

so the used code is wrong in many ways.